packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2127,6 +2147,10 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           this.unexpected();
         });
 
+        if (!result && isOptionalCall) {
+          this.unexpected(this.state.pos, tt.parenL);
```

Comment:
I think we can throw at better position: We can track the position after `typeArguments` is parsed but we do not see a left parenthesis token. Then we throw on this position that we are expecting a left paren _after_ type arguments.

Reply:
If we throw an error on this condition, the `result` would be `undefined` by `tsTryParseAndCatch`, and also the error message would be ignored.
I found that the error position is wrong (it should be "SyntaxError: Unexpected token, expected \"(\" (1:12)") because the error has been restored by `tsTryParseAndCatch`. 
I wanna throw an error with the accurate error message at a better position. 
To handle this, we can make `unexpectedError` in `packages/babel-parser/src/parser/util.js` and return `this.unexpectedError(this.state.pos, tt.parenL)` on that condition., then change this part like this. 
```
if (result?.failState) return result.node;
```
But I'm not sure that we could use `node` to handle the error. It seems weird. 😢

Reply:
I patched the code to help you understanding.